src/dbally/views/freeform/text2sql/_view.py

```diff
@@ -91,22 +154,28 @@ async def ask(
 
     async def _generate_sql(
```

Comment:
@ludwiktrammer wdyt about introducing strategy pattern here?

Initially we would have 2 strategies:
* in case there is no similarity indexes defined we use prompt without "parameters" logic (to reduce token usage)
* with indexes defined we use parameter binding

Reply:
Maybe, although to be honest I like the decoupling of user-provided values and the SQL code as a general idea, even without similarity indexes. It seems to me that it makes SQL injections less likely (although still possible) and gives us more visibility.